core/fedavgoptimizer.py

```diff
@@ -0,0 +1,144 @@
+from core.utils.event_types import ActionableEventTypes, RawEventTypes, callback_handler
+from core.utils.dmljob import deserialize_job
+from core.utils.dmlresult import DMLResult
+
+class FederatedAveragingOptimizer(object):
+	"""
+	Federated Averaging Optimizer
+
+	Implements the Federated Averaging algorithm (with some tweaks)
+	and acts as a "brain" for the Unix Service
+
+	"""
+	def __init__(self, content):
+		"""
+		Kwargs is a dictionary containing all the info needed to initialize the Optimizer.
+		This includes any number of properties such as thresholds, etc.
+		"""
```

Comment:
Can you add start and finished logging to the init? Just like the Communication Manager does it.

Reply:
done